### PR TITLE
[SPARK-13387][MESOS] Add support for SPARK_DAEMON_JAVA_OPTS with MesosClusterDispatcher.

### PR DESCRIPTION
**Pull request by tnachen**
## What changes were proposed in this pull request?

Add support for SPARK_DAEMON_JAVA_OPTS with MesosClusterDispatcher.
## How was the this patch tested?

Manual testing by launching dispatcher with SPARK_DAEMON_JAVA_OPTS

### CODE REVIEW

**Comment by tnachen** [comment]
@andrewor14 @dragos It's a pretty small change, PTAL

**Comment by andrewor14** [comment]
This looks OK, but can you elaborate on the motivation?

File: launcher/src/main/java/org/apache/spark/launcher/SparkClassCommandBuilder.java [hunk 1]
```diff
@@ -48,8 +48,8 @@
     String memKey = null;
     String extraClassPath = null;
 
-    // Master, Worker, and HistoryServer use SPARK_DAEMON_JAVA_OPTS (and specific opts) +
-    // SPARK_DAEMON_MEMORY.
+    // Master, Worker, HistoryServer, ExternalShuffleService, MesosDispatcher use
```
Comment (andrewor14) [inline launcher/src/main/java/org/apache/spark/launcher/SparkClassCommandBuilder.java]: MesosClusterDispatcher

**Comment by tnachen** [comment]
@andrewor14 This is so that users can be able to set java properties when running the mesos cluster dispatcher, it's really useful for testing the dispatcher especially since in the integration test suite we can more easily override certain property values without changing spark configuration files. This is going to also be easier for people running dispatchers to change configurations when bootstrapping it more easier.

**Comment by tnachen** [comment]
@andrewor14 Fixed your comment now, PTAL

**Comment by dragos** [comment]
`SPARK_DEAMON_JAVA_OPTS` seems to be understood by the other daemons, so LGTM!

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
Actually this already passed tests, merging into master.

**Pull request closed without merging**